src/chart/generateCategoricalChart.tsx

```diff
@@ -233,7 +233,7 @@ const getTooltipContent = (
      * Defaulting to chartData below to fix an edge case where the tooltip does not include data from all charts
      * when a separate dataset is passed to chart prop data and specified on Line/Area/etc prop data
      */
-    const data = (child.props.data ?? chartData).slice(state.dataStartIndex, state.dataEndIndex + 1);
+    const data = (child.props.data ?? chartData)?.slice(state.dataStartIndex, state.dataEndIndex + 1);
```

Comment:
Could you please add a test to prevent the bug from coming up again?

Reply:
Testing this function is not easy, i wating for a reproducible example.
https://github.com/recharts/recharts/pull/3932#issuecomment-1809236675

Reply:
Could we extract this function as an independent helper function? Possibly a follow up task.